src/test/compile-fail/borrowck/borrowck-local-borrow-outlives-fn.rs

```diff
@@ -0,0 +1,28 @@
+// Copyright 2016 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+// revisions: ast mir
+//[mir]compile-flags: -Z emit-end-regions -Z borrowck-mir
+
+fn cplusplus_mode(x: isize) -> &'static isize {
+    &x //[ast]~ ERROR `x` does not live long enough
+       //[mir]~^ ERROR `x` does not live long enough (Ast)
+       //[mir]~| ERROR borrowed value does not live long enough (Mir)
+}
+
+fn cplusplus_mode_exceptionally_unsafe(x: &mut Option<&'static mut isize>) {
+    let mut z = 0;
+    *x = Some(&mut z); //[ast]~ ERROR `z` does not live long enough
+                       //[mir]~^ ERROR `z` does not live long enough (Ast)
+                       //[mir]~| ERROR `z` does not live long enough (Mir)
```

Comment:
The only problem left here now is this - MIR borrowck doesn't generate an error for this.

Reply:
The problem here is `TerminatorKind::Call` with a `None` unwind terminator, which just means that the call immediately jumps to a resume.

I feel it's a good idea to have MIR construction instead link all `None` call terminators to a resume block, and have a later pass (e.g. `SimplifyBranches`) clean up unwind paths that don't actually do anything so they won't choke LLVM.

Reply:
I suppose you can remove this test and I'll handle the change myself.